lk2nd/device/device.c

```diff
@@ -126,6 +127,13 @@ static void parse_dtb(const void *dtb)
 	if (len >= 0)
 		lk2nd_dev.single_key = true;
 
+	strings = (const char *const *)
+		lkfdt_stringlist_get_all(dtb, node, "lk2nd,menu-key-strings", &len);
```

Comment:
This function allocates an array and strdups strings into it already, in this case you will leak them all since you don't save the whole array as done above. First thought would be to copy pointers from the array as-is and free the outer array then but you'd need to handle extra values if len>2 then...

I'd say here you should use something like `blah = strdup(fdt_stringlist_get(...))` and it seem to return NULL if the prop doesn't exist or doesn't have a string of given index